pkg/receive/handler.go

```diff
@@ -95,9 +87,12 @@ func NewHandler(logger log.Logger, o *Options) *Handler {
 	return h
 }
 
-// StorageReady marks the storage as ready.
-func (h *Handler) StorageReady() {
-	atomic.StoreUint32(&h.storageReady, 1)
+// Storage sets the writer and marks the storage as ready.
+// If the writer is nil, then the storage is marked as not ready.
+func (h *Handler) Storage(w *Writer) {
```

Comment:
Shouldn't this be called "SetWriter"?

and just comment that it needs to be called before using the endpoint

Reply:
naming is hard!